src/main/java/org/roaringbitmap/RunContainer.java

```diff
@@ -942,6 +932,40 @@ public boolean equals(Object o) {
     return false;
   }
 
+  private boolean equals(RunContainer runContainer) {
+    if (runContainer.nbrruns != this.nbrruns) {
+      return false;
+    }
+    for (int i = 0; i < nbrruns; ++i) {
+      if (getValue(i) != runContainer.getValue(i)) {
+        return false;
+      }
+      if (getLength(i) != runContainer.getLength(i)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  private boolean equals(ArrayContainer arrayContainer) {
+    if (getCardinality() != arrayContainer.getCardinality()) {
+      return false;
+    }
+    int pos = 0;
+    for (short i = 0; i < nbrruns; ++i) {
+      short runStart = getValue(i);
+      short length = getLength(i);
+      if (arrayContainer.content[pos] != runStart) {
+        return false;
+      }
+      if (arrayContainer.content[pos + length] != runStart + length) {
+        return false;
+      }
+      pos += length + 1;
+    }
+    return pos == arrayContainer.getCardinality();
```

Comment:
Algorithmically, no. I thought it was a nice way of expressing/reinforcing the loop invariant. We can just return true if you prefer...